lib/lita/handlers/reviewme.rb

```diff
@@ -86,28 +88,20 @@ def generate_assignment(response, room: get_room(response))
         response.reply(reviewer.to_s)
       end
 
-      def comment_on_github(response, room: get_room(response))
-        repo = response.match_data[:repo]
-        id = response.match_data[:id]
+      def review_on_github(response, room: get_room(response))
+        github = Lita::Reviewme::Github.new(config, response.match_data[:repo], response.match_data[:id])
 
-        reviewer = next_reviewer(room)
-        begin
-          pull_request = github_client.pull_request(repo, id)
-          owner = pull_request.user.login
-          reviewer = next_reviewer(room) if owner == reviewer
-        rescue Octokit::Error
-          response.reply("Unable to check who issued the pull request. Sorry if you end up being assigned your own PR!")
-        end
-        return response.reply('Sorry, no reviewers found') unless reviewer
-        comment = github_comment(reviewer, pull_request)
-
-        begin
-          github_client.add_comment(repo, id, comment)
+        if reviewer = next_reviewer(room, github.owner)
+          github.assign(reviewer)
           response.reply("#{reviewer} should be on it...")
-        rescue Octokit::Error
-          url = response.match_data[:url]
-          response.reply("I couldn't post a comment. (Are the permissions right?) #{chat_mention(reviewer, url)}")
+        else
+          response.reply('Sorry, no reviewers found')
         end
+      rescue Lita::Reviewme::Github::UnknownOwner
+        response.reply("Unable to check who issued the pull request. Sorry if you end up being assigned your own PR!")
+      rescue Lita::Reviewme::Github::CannotPostComment
+        url = response.match_data[:url]
+        response.reply("I couldn't post a comment or request a reviewer. (Are the permissions right?) #{chat_mention(reviewer, url)}")
```

Comment:
This is where my question about the error comes up again. We might see this problem if the lib is misconfigured to never assign in any way (if I'm understanding the above correctly). So we might want to handle the other scenario separately so we can communicate something like "Whoops! The lib is configured to not assign in any way". It actually might make sense to check for that scenario in this object since it owns those configuration values. That is, not make it the responsibility of the Github object to enforce.